operator/src/main/java/oracle/kubernetes/operator/helpers/JobHelper.java

```diff
@@ -303,64 +303,115 @@ public NextAction apply(Packet packet) {
-      cleanupJobArtifacts(packet);
-      return super.onFailure(packet, callResponse);
-    }
-
     @Override
     public NextAction onSuccess(Packet packet, CallResponse<String> callResponse) {
       String result = callResponse.getResult();
-
-      // Log output to Operator log
       LOGGER.fine("+++++ ReadDomainIntrospectorPodLogResponseStep: \n" + result);
 
-      // Parse out each job log message and log to Operator as SEVERE, FINE, etc...
       if (result != null) {
         convertJobLogsToOperatorLogs(result);
+        if (!severeStatuses.isEmpty()) updateStatus(packet.getSpi(DomainPresenceInfo.class));
+        packet.put(ProcessingConstants.DOMAIN_INTROSPECTOR_LOG_RESULT, result);
       }
 
-      V1Job domainIntrospectorJob = (V1Job) packet.get(ProcessingConstants.DOMAIN_INTROSPECTOR_JOB);
-      if (domainIntrospectorJob != null && JobWatcher.isComplete(domainIntrospectorJob)) {
-        if (result != null) {
-          packet.put(ProcessingConstants.DOMAIN_INTROSPECTOR_LOG_RESULT, result);
-        }
+      V1Job domainIntrospectorJob =
+            (V1Job) packet.remove(ProcessingConstants.DOMAIN_INTROSPECTOR_JOB);
+      if (isNotComplete(domainIntrospectorJob)) return onFailure(packet, callResponse);
 
-        // Delete the job once we've successfully read the result
-        DomainPresenceInfo info = packet.getSpi(DomainPresenceInfo.class);
-        java.lang.String domainUid = info.getDomain().getDomainUid();
-        java.lang.String namespace = info.getNamespace();
+      return doNext(packet);
+    }
 
-        cleanupJobArtifacts(packet);
+    private boolean isNotComplete(V1Job domainIntrospectorJob) {
+      return domainIntrospectorJob == null || !JobWatcher.isComplete(domainIntrospectorJob);
+    }
 
-        return doNext(
-            JobHelper.deleteDomainIntrospectorJobStep(domainUid, namespace, getNext()), packet);
+    // Parse log messages out of a Job Log
+    //  - assumes each job log message starts with '@['
+    //  - assumes any lines that don't start with '@[' are part
+    //    of the previous log message
+    //  - ignores all lines in the log up to the first line that starts with '@['
+    private void convertJobLogsToOperatorLogs(String jobLogs) {
+      for (String line : jobLogs.split(EOL_PATTERN)) {
+        if (line.startsWith("@[")) {
+          logToOperator();
+          logMessage = new StringBuilder(INTROSPECTOR_LOG_PREFIX).append(line.trim());
+        } else if (logMessage.length() > 0) {
+          logMessage.append(System.lineSeparator()).append(line.trim());
+        }
+      }
+      logToOperator();
+    }
+
+    private void logToOperator() {
+      if (logMessage.length() == 0) return;
+
+      String logMsg = logMessage.toString();
+      switch (getLogLevel(logMsg)) {
+        case "SEVERE":
+          addSevereStatus(logMsg); // fall through
+        case "ERROR":
+          LOGGER.severe(logMsg);
```

Comment:
I realize this pull has already been merged, but ERROR should also cause an 'addSevereStatus' since an ERROR is converted to a severe.   (I think I removed all errors, so this is not strictly needed - but this path exists just in case someone adds one back in later.)

Reply:
If you do this, then you'll need to search for last-index-of ERROR in addition to SEVERE in the 'addSevereStatus'  method.